File: wrapper/csharp/src/bq/category_log.cs
/*
 * Copyright (C) 2025 Tencent.
 * BQLOG is licensed under the Apache License, Version 2.0.
 * You may obtain a copy of the License at
 * 
 *     http://www.apache.org/licenses/LICENSE-2.0
 * 
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 */
using System.Collections.Generic;

namespace bq
{
    public class category_log : log
    {
        protected category_log() : base()
        {
        }

        /// <summary>
        /// copy constructor
        /// </summary>
        /// <param name="rhs"></param>
        public category_log(log child_inst) : base(child_inst) 
        {
        }

        /// <summary>
        /// Get log categories count
        /// </summary>
        /// <returns></returns>
        public uint get_categories_count()
        {
            return (uint)categories_name_array_.Count;
        }

        /// <summary>
        /// Get names of all categories
        /// </summary>
        /// <returns></returns>
        public List<string> get_categories_name_array()
        {
            return categories_name_array_;
        }
    }
}

File: wrapper/go/src/bq/category_log.go
/* Copyright (C) 2026 Tencent.
 * BQLOG is licensed under the Apache License, Version 2.0.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 */
package bq

import (
	"github.com/Tencent/BqLog/wrapper/go/src/bq/def"
)

// Category_log attaches a category index to each entry; categories are the
// strings passed to Create_log.
type Category_log struct {
	*Log
}

func Create_category_log(name, config string, category_names []string) *Category_log {
	l := Create_log(name, config, category_names)
	return &Category_log{Log: l}
}

func (l *Category_log) Write_c(level def.Log_level, category_index uint32, format string, args ...Arg) bool {
	return l.Write(level, category_index, format, args...)
}

func (l *Category_log) Verbose_c(category_index uint32, format string, args ...Arg) bool {
	return l.Write(def.Verbose, category_index, format, args...)
}
func (l *Category_log) Debug_c(category_index uint32, format string, args ...Arg) bool {
	return l.Write(def.Debug, category_index, format, args...)
}
func (l *Category_log) Info_c(category_index uint32, format string, args ...Arg) bool {
	return l.Write(def.Info, category_index, format, args...)
}
func (l *Category_log) Warning_c(category_index uint32, format string, args ...Arg) bool {
	return l.Write(def.Warning, category_index, format, args...)
}
func (l *Category_log) Error_c(category_index uint32, format string, args ...Arg) bool {
	return l.Write(def.Error, category_index, format, args...)
}
func (l *Category_log) Fatal_c(category_index uint32, format string, args ...Arg) bool {
	return l.Write(def.Fatal, category_index, format, args...)
}
